test/example/start

```diff
@@ -15,3 +15,5 @@ cluster = sys.argv[1]
 
 print("Deploying example")
 kubectl.apply("--filename", "example/deployment.yaml", context=cluster)
+print("Creating ingress")
+kubectl.apply("--filename", "example/ingress.yaml")
```

Comment:
Missing context=cluster - this will install on the currently current context - it may work by accident but fail later. Always include a context!

Reply:
Another issue - you never wait until the ingress is ready - see how we wait for the deployment in example/test.

But what we should really do is to wait for the deployment in `example/start`, like other components, and in `example/test`, we want to have a test verifying the deployment. Since we installed ngix and ingress, the web server should be availale
and we should be able to run curl command to download a default html or something
similar.

Reply:
Regarding missing context - in this case it will always not work - because the start
scripts runs with the cluster name:

    example/start ex1
    example/start ex2

The current code will always apply the yaml to the same cluster, then one that happened to be defined as current context at the time of the call.

Reply:
Regarding waiting - you probably need to wait until ngix is ready
before you configure ingress to expose it. Check the docs to see
if this is the case.

I get this error after moving to kvm2 driver and adding the missing context:

```
2023-03-19 18:10:15,352 ERROR   [ex1/0] Cluster failed
Traceback (most recent call last):
  File "/home/nsoffer/src/ramen/test/drenv/__main__.py", line 111, in execute
    f.result()
  File "/usr/lib64/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/usr/lib64/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nsoffer/src/ramen/test/drenv/__main__.py", line 264, in run_worker
    run_script(script, worker["name"], hooks=hooks, allow_failure=allow_failure)
  File "/home/nsoffer/src/ramen/test/drenv/__main__.py", line 271, in run_script
    run_hook(hook, script["args"], name, allow_failure=allow_failure)
  File "/home/nsoffer/src/ramen/test/drenv/__main__.py", line 278, in run_hook
    run(hook, *args, name=name)
  File "/home/nsoffer/src/ramen/test/drenv/__main__.py", line 293, in run
    for line in commands.watch(*cmd):
  File "/home/nsoffer/src/ramen/test/drenv/commands.py", line 127, in watch
    raise Error(args, p.returncode, error)
drenv.commands.Error: Command failed:
   command: ('example/start', 'ex1')
   exitcode: 1
   error:
      Traceback (most recent call last):
        File "/home/nsoffer/src/ramen/test/example/start", line 19, in <module>
          kubectl.apply("--filename", "example/ingress.yaml", context=cluster)
        File "/home/nsoffer/src/ramen/test/drenv/kubectl.py", line 41, in apply
          _watch("apply", *args, input=input, context=context)
        File "/home/nsoffer/src/ramen/test/drenv/kubectl.py", line 95, in _watch
          for line in commands.watch(*cmd, input=input):
        File "/home/nsoffer/src/ramen/test/drenv/commands.py", line 127, in watch
          raise Error(args, p.returncode, error)
      drenv.commands.Error: Command failed:
         command: ('kubectl', 'apply', '--context', 'ex1', '--filename', 'example/ingress.yaml')
         exitcode: 1
         error:
            Error from server (InternalError): error when creating "example/ingress.yaml": Internal error occurred: failed calling webhook "validate.nginx.ingress.kubernetes.io": failed to call webhook: Post "https://ingress-nginx-controller-admission.ingress-nginx.svc:443/networking/v1/ingresses?timeout=10s": dial tcp 10.107.79.220:443: connect: connection refused
```

But when running `example/start` again manually:

```
$ example/start ex1
Deploying example
service/nginxservice unchanged
deployment.apps/example-deployment unchanged
Creating ingress
ingress.networking.k8s.io/ingress created

$ example/start ex2
Deploying example
service/nginxservice unchanged
deployment.apps/example-deployment unchanged
Creating ingress
ingress.networking.k8s.io/ingress created
```

Looking at the docs, we need to wait until the ingress controller is ready:
https://kubernetes.io/docs/tasks/access-application-cluster/ingress-minikube/
(See "2. Verify that the NGINX Ingress controller is running")

Looking at the deployments:
```
$ kubectl get deploy -A --context ex1
NAMESPACE       NAME                       READY   UP-TO-DATE   AVAILABLE   AGE
default         example-deployment         1/1     1            1           10m
ingress-nginx   ingress-nginx-controller   1/1     1            1           11m
kube-system     coredns                    1/1     1            1           11m
```

We likely need to wait until ingress-nginx-controller is rolled out before adding the ingress.

I tried this patch on top of your pr:

```diff
diff --git a/test/example.yaml b/test/example.yaml
index d4e8c6d..c09897c 100644
--- a/test/example.yaml
+++ b/test/example.yaml
@@ -4,13 +4,15 @@
 # Example environment.
 ---
 name: example
 templates:
   - name: "example-cluster"
-    driver: podman
-    container_runtime: cri-o
-    addons: 'ingress'
+    driver: kvm2
+    container_runtime: containerd
+    network: default
+    addons:
+      - ingress
     workers:
       - scripts:
           - name: example
 profiles:
   - name: ex1
diff --git a/test/example/ingress.yaml b/test/example/ingress.yaml
index b45e8d0..5532009 100644
--- a/test/example/ingress.yaml
+++ b/test/example/ingress.yaml
@@ -2,11 +2,11 @@
 # SPDX-License-Identifier: Apache-2.0
 ---
 apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
-  name: ingress
+  name: example-ingress
 spec:
   ingressClassName: nginx-example
   rules:
   - host: localhost
     http:
diff --git a/test/example/start b/test/example/start
index d814325..0eb6f17 100755
--- a/test/example/start
+++ b/test/example/start
@@ -13,7 +13,28 @@ if len(sys.argv) != 2:
 
 cluster = sys.argv[1]
 
 print("Deploying example")
 kubectl.apply("--filename", "example/deployment.yaml", context=cluster)
+
+print("Waiting until deployment is rolled out")
+kubectl.rollout(
+    "status",
+    "deploy/example-deployment",
+    "--timeout=180s",
+    context=cluster,
+)
+
+print("Waiting until ingress controller deployment is rolled out")
+kubectl.rollout(
+    "status",
+    "deploy/ingress-nginx-controller",
+    "--namespace=ingress-nginx",
+    "--timeout=180s",
+    context=cluster,
+)
+
 print("Creating ingress")
-kubectl.apply("--filename", "example/ingress.yaml")
+kubectl.apply("--filename", "example/ingress.yaml", context=cluster)
+
+print("Waiting until ingress is ready")
+# XXX how to do this?
diff --git a/test/example/test b/test/example/test
index 1b7c959..7cd0824 100755
--- a/test/example/test
+++ b/test/example/test
@@ -12,11 +12,6 @@ if len(sys.argv) != 2:
     sys.exit(1)
 
 cluster = sys.argv[1]
 
 print("Testing example deployment")
-kubectl.rollout(
-    "status",
-    "deploy/example-deployment",
-    "--timeout=180s",
-    context=cluster,
-)
+# XXX how to do that?
```

I did not test the ingress or accessing nginx, but the environment starts
without errors.

```
$ drenv start example.yaml 
2023-03-19 18:35:48,963 INFO    [example] Starting environment
2023-03-19 18:35:49,036 INFO    [ex1] Starting minikube cluster
2023-03-19 18:35:50,063 INFO    [ex2] Starting minikube cluster
2023-03-19 18:37:35,759 INFO    [ex1] Cluster started in 106.72 seconds
2023-03-19 18:37:35,760 INFO    [ex1/0] Running example/start
2023-03-19 18:37:51,792 INFO    [ex2] Cluster started in 121.73 seconds
2023-03-19 18:37:51,792 INFO    [ex2/0] Running example/start
2023-03-19 18:37:52,600 INFO    [ex1/0] example/start completed in 16.84 seconds
2023-03-19 18:37:52,600 INFO    [ex1/0] Running example/test
2023-03-19 18:37:52,634 INFO    [ex1/0] example/test completed in 0.03 seconds
2023-03-19 18:38:10,114 INFO    [ex2/0] example/start completed in 18.32 seconds
2023-03-19 18:38:10,114 INFO    [ex2/0] Running example/test
2023-03-19 18:38:10,138 INFO    [ex2/0] example/test completed in 0.02 seconds
2023-03-19 18:38:10,139 INFO    [example] Environment started in 141.18 seconds
```